client/components/app.js

```diff
@@ -0,0 +1,119 @@
+    document.getElementById('display-none').id = newButtonId;
+    target.id = "display-none";
+    this.setState((state) => ({ playClicked: !state.playClicked }));
+  }
+  startPlayTimer() {
+    const intervalId = setInterval((() => {
+      this.setState((state) => { return { currentTimestamp: (state.currentTimestamp + 1) } });
+    }).bind(this), 1000);
+    this.setState({ currentIntervalId: intervalId });
+  }
+  stopPlayTimer() {
+    clearInterval(this.state.currentIntervalId);
+  }
+  goToArtistPage({ target }) {
+    alert(`This would send you to the profile page for ${target.innerHTML}`);
+  }
+  goToPopularTracksForTag({ target }) {
+    alert(`This would send you to the Most Popular Tracks for ${target.innerHTML} page`);
+  }
+  changePostDateToString(date) {
+    const releaseText = {
+      0: ' minutes ago',
+      1: ' hours ago',
+      2: ' days ago',
+      3: ' weeks ago',
+      4: ' months ago',
+      5: ' years ago',
+    };
+    const releaseTime = [];
+    const parsedDate = Date.parse(date);
+    const timeAgoInMilliseconds = (Date.now() - parsedDate);
+    const timeAgoInMinutes = timeAgoInMilliseconds/1000/60;
+    releaseTime.push(timeAgoInMinutes);
+    const timeAgoInHours = Math.floor(timeAgoInMinutes/60);
+    releaseTime.push(timeAgoInHours);
+    const timeAgoInDays = Math.floor(timeAgoInHours/24);
+    releaseTime.push(timeAgoInDays);
+    const timeAgoInWeeks = Math.floor(timeAgoInDays/7);
+    releaseTime.push(timeAgoInWeeks);
+    const timeAgoInMonths = Math.floor(timeAgoInWeeks/4);
+    releaseTime.push(timeAgoInMonths);
+    const timeAgoInYears = Math.floor(timeAgoInMonths/12);
+    releaseTime.push(timeAgoInYears);
+
+    for (let i = releaseTime.length - 1; i >= 0; i -= 1) {
+      if (releaseTime[i] > 0) return (releaseTime[i].toString()).concat(releaseText[i]);
+    }
+  }
+  popoutArtworkWindow({ target }) {
+    // console.dir(document.getElementById('app'));
+    // document.getElementById('app').style.setProperty('filter', 'grayscale(100%)');
+    target.nextSibling.classList.remove('hidden');
+  }
+  closeArtworkWindow({ target }) {
+    // console.dir(target);
+    // document.getElementById('app').style.setProperty('filter', 'none');
+    target.parentNode.classList.add('hidden');
+  }
+
+  render() {
+    return (
+      <>
+        <div id="display-play-button" className="play-pause-button" onClick={(event) => {
+          this.changePlayPauseButton(event);
+          this.startPlayTimer();
+        }}></div>
+        <div id="display-none" className="play-pause-button" onClick={(event) => {
+          this.changePlayPauseButton(event);
+          this.stopPlayTimer();
+        }}></div>
+        <table id="song-info-table">
+          <tbody>
+            <tr>
+              <td><div id="artist" className="text" onClick={(event) => this.goToArtistPage(event)}>{this.props.songData.artist}</div></td>
```

Comment:
I don't think Aruna or Rahim used id's so I doubt there will be conflicts.  However, some of the classes like play, comment, etc: You may want to add some unique identifier like "Display" to the end to make sure the class is unique to your component.

Reply:
Noted. I will keep this in mind once we all start combining things.